### PR TITLE
HW-53656 create processing time class

### PR DESCRIPTION
**Pull request by vshcherbyna-epam**
HW-53655: Create ProcessingTimeTest class

### CODE REVIEW

File: core/src/main/java/com/hyperwallet/android/model/graphql/ProcessingTime.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                : that.mTransferMethodType == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = mCountry != null ? mCountry.hashCode() : 0;
+        result = 31 * result + (mCurrency != null ? mCurrency.hashCode() : 0);
+        result = 31 * result + (mTransferMethodType != null ? mTransferMethodType.hashCode() : 0);
+        return result;
+    }
```
Comment (azakrevska-epam) [inline core/src/main/java/com/hyperwallet/android/model/graphql/ProcessingTime.java]: @vshcherbyna-epam maybe we can use Objects.hash ?

Comment (vshcherbyna-epam) [reply core/src/main/java/com/hyperwallet/android/model/graphql/ProcessingTime.java]: Yeah, thanks, it is good utility.

File: core/src/main/java/com/hyperwallet/android/model/graphql/ProcessingTime.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    private static final String COUNTRY = "country";
+    private static final String CURRENCY = "currency";
+    private static final String TRANSFER_METHOD_TYPE = "transferMethodType";
+    private static final String VALUE = "value";
+    private final String mCountry;
+    private final String mCurrency;
+    private final String mTransferMethodType;
+    private final String mValue;
+
+    public ProcessingTime(JSONObject jsonObject) {
```
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/model/graphql/ProcessingTime.java]: Please add final and @NonNull if applicable

File: core/src/test/java/com/hyperwallet/android/model/graphql/ProcessingTimeTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                "processing_time_response.json");
+        JSONObject jsonResponseObject = new JSONObject(jsonResponse);
+        ProcessingTime actualProcessingTime = new ProcessingTime(jsonResponseObject);
+        ProcessingTime expectedProcessingTime = new ProcessingTime(new JSONObject("{\n"
+                + "  \"transferMethodType\": \"BANK_CARD\",\n"
+                + "  \"country\": \"CA\",\n"
+                + "  \"currency\": \"CAD\",\n"
+                + "  \"value\": \"1-2\"\n"
+                + "}"));
+        assertEquals(actualProcessingTime, expectedProcessingTime);
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/graphql/ProcessingTimeTest.java]: please use assertThat

File: core/src/test/java/com/hyperwallet/android/model/graphql/ProcessingTimeTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        ProcessingTime processingTime = new ProcessingTime(jsonResponseObject);
+
+        assertThat(processingTime.getTransferMethodType(), is(BANK_CARD));
+        assertThat(processingTime.getCountry(), is("CA"));
+        assertThat(processingTime.getCurrency(), is("CAD"));
+        assertThat(processingTime.getValue(), is("1-3 Business days"));
+    }
+
+    @Test
+    public void testProcessingTime_equalsHashCode() throws JSONException {
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/graphql/ProcessingTimeTest.java]: unless this test is only required to achieve the minimum coverage ratio, this test can be deleted

Comment (vshcherbyna-epam) [reply core/src/test/java/com/hyperwallet/android/model/graphql/ProcessingTimeTest.java]: @fmattos-hw due to we override default behavior of Object.equls and Object.hashCode I just wanted to keep contract of them.

**Pull request merged**